aion/lib/aion/room_channel/room.ex

```diff
@@ -73,15 +87,33 @@ defmodule Aion.RoomChannel.Room do
 
   @spec get_current_question(__MODULE__.t) :: Question.t
   def get_current_question(room) do
-    room.question
+    room.current_question
   end
 
-  @spec get_new_question_with_answers(integer) :: %{question: Question.t, answers: list Answer.t}
-  defp get_new_question_with_answers(room_id) do
-      question = Question.get_random_question(room_id)
-      answers = Answer.get_answers(question.id)
-      Logger.debug fn -> "Answers: #{inspect(Enum.map(answers, fn answer -> answer.content end))}" end
+  @spec get_new_question_with_answers((list Question.t), integer) :: new_question_set
+  defp get_new_question_with_answers(questions, room_id) do
+    case questions do
+      [last_question] ->
+        remaining_questions = fetch_questions(room_id)
+        get_new_question_set(last_question, remaining_questions)
+
+      [next_question | remaining_questions] ->
+        get_new_question_set(next_question, remaining_questions)
+
+      [] ->
```

Comment:
You could move that as the first clause of the pattern match. It's easier and more intuitive to read the pattern matches when they are ordered in a way. Like here it would be:
```
case list of
    empty_list -> 
    list_with_one_elem ->
    list_with_more -> 
end
```